### PR TITLE
Clean up and refactor code

### LINKED ISSUE #36: Clean up and refactoring of code

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**
- Based on review left at https://www.notion.so/avantifellows/Code-Review-ff6484998d1e4ef4b697ff4c02d57b6e.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,5 +1,7 @@
-VUE_APP_AF_API_KEY=''
+VUE_APP_API_KEY=''
 VUE_APP_BASE_URL_PLIO=''
 VUE_APP_AWS_SQS_ACCESS_KEY=''
-VUE_APP_AWS_SQS_SECRET_ACCESS_KEY=''
+VUE_APP_AWS_SQS_SECRET_KEY=''
 VUE_APP_BASE_URL_MEET=''
+VUE_APP_AWS_REGION=''
+VUE_APP_AWS_SQS_URL=''
```
Comment (deepansh96) [inline .env.example]: Looks good!
Can we do some organizing here? Like a comment like "# aws" and all the things related to that go under that comment, and similarly for others. For example, right now we have two env variables here for redirection targets, so maybe we can add that section as well (just as a comment)
What do you think?

File: docs/DEPLOYMENT.MD [hunk 2]
```diff
@@ -17,5 +16,7 @@ This document covers steps on setting up this repository on various cloud hostin
      ```
 8. To deploy your code on to staging, use `npm run deploy-staging/yarn deploy-staging`.
 
-# Production
-Setting up a production environment on AWS is same as staging.
+### Production
+Setting up a production environment on AWS is the same as setting up a staging environment.
+
+**Make sure to setup `.env.staging` for staging environment and `.env.production` for production environment.**
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD]: I think the script for `deploy` should be in this section and the staging section should just contain the script for `deploy-staging` no?

File: docs/ENV.MD [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
-## VUE_APP_BASE_URL_PLIO=''
+## VUE_APP_BASE_URL_PLIO
 Base URL for the Plio link.
 
-## VUE_APP_AWS_SQS_ACCESS_KEY=''
+## VUE_APP_AWS_SQS_ACCESS_KEY
 AWS access key for access to AWS SQS.
 
-## VUE_APP_AWS_SQS_SECRET_ACCESS_KEY=''
+## VUE_APP_AWS_SQS_SECRET_ACCESS_KEY
```
Comment (deepansh96) [inline docs/ENV.MD]:
```suggestion
## VUE_APP_AWS_SQS_SECRET_KEY
```

File: package.json [hunk 4]
```diff
@@ -16,6 +16,7 @@
     "core-js": "^3.6.5",
     "dns": "^0.2.2",
     "postcss": "^7.0.36",
+    "pre-commit": "^1.2.2",
```
Comment (deepansh96) [inline package.json:19]: What was the need for adding this?

File: src/components/SRNEntry.vue [hunk 5]
```diff
@@ -100,35 +103,35 @@ export default {
...
-        this.userIdListLength > 0 &&
+        this.numOfUserIds > 0 &&
         this.userIDList[0]["userID"] != ""
       );
     },
     /** Whether multiple entries have been made by the user */
-    ifUserEnteredMoreThanOne() {
-      return !this.isSingleEntryOnly && this.userIdListLength > 1;
+    hasUserEnteredMoreThanOne() {
+      return !this.isMultipleIDEntryAllowedEntryOnly && this.numOfUserIds > 1;
```
Comment (deepansh96) [inline src/components/SRNEntry.vue]: This name seems confusing to me i think! If anyone else also thinks so maybe can you think of an alternative?

File: src/main.js [hunk 6]
```diff
@@ -1,11 +1,10 @@
 import { createApp } from 'vue'
 import App from './App.vue'
 import router from "./router";
-import store from "./store";
 import './index.css';
 import InlineSvg from "vue-inline-svg";
 
-const app = createApp(App).component('inline-svg', InlineSvg).use(store).use(router);
+const app = createApp(App).component('inline-svg', InlineSvg).use(router);
```
Comment (deepansh96) [inline src/main.js]: any reason for removing store?

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: src/main.js (hunk 6 above)
Comment (sudheshna-donthineni) [reply src/main.js]: I thought we discussed on the call that we can bring back store later when we start adding dropdowns.

File: src/components/SRNEntry.vue (hunk 5 above)
Comment (sudheshna-donthineni) [reply src/components/SRNEntry.vue]: Sorry, that was a typo!

File: .env.example (hunk 1 above)
Comment (sudheshna-donthineni) [reply .env.example]: I'm not sure I understand this comment. The comments on env variables are in the docs.

Comment (deepansh96) [reply .env.example]: Yeah I explained it pretty bad 🤦 
We'll discuss on call

File: src/main.js (hunk 6 above)
Comment (deepansh96) [reply src/main.js]: Ohhh I forgot it guess
cool cool

File: package.json (hunk 4 above)
Comment (sudheshna-donthineni) [reply package.json:19]: Pre-commit wasn't added to this project locally.

File: docs/DEPLOYMENT.MD (hunk 2 above)
Comment (sudheshna-donthineni) [reply docs/DEPLOYMENT.MD]: Pritam's making some changes to the documentation.

Comment (pritamps) [reply docs/DEPLOYMENT.MD]: I'm just correcting stuff lol. Don't even know what @deepansh96 is talking about, so please make the changes he's suggesting

File: .env.example [hunk 7]
```diff
@@ -1,5 +1,14 @@
-VUE_APP_AF_API_KEY=''
-VUE_APP_BASE_URL_PLIO=''
+### .env variables for setting up AWS SQS
```
Comment (deepansh96) [inline .env.example]: I feel `.env variables for setting up` is redundant
Because this file is called `.env` so it's implied in a way. What do you think?

File: .env.example [hunk 8]
```diff
@@ -1,5 +1,14 @@
...
 VUE_APP_AWS_SQS_ACCESS_KEY=''
-VUE_APP_AWS_SQS_SECRET_ACCESS_KEY=''
+VUE_APP_AWS_SQS_SECRET_KEY=''
+VUE_APP_AWS_REGION=''
+VUE_APP_AWS_SQS_URL=''
+
+### .env variables for supporting different destinations
+VUE_APP_API_KEY=''
```
Comment (deepansh96) [inline .env.example:8]: What was this for? I totally forgot 😕

File: .github/workflows/deploy_to_s3_prod.yml [hunk 9]
```diff
@@ -26,19 +26,24 @@ jobs:
...
     - name: Install
       run: npm install
-    
+
     # build the vuepress docs
     - name: Build
       env:
         NODE_ENV: production
-        VUE_APP_AF_API_KEY: ${{ secrets.VUE_APP_AF_API_KEY }}
+        VUE_APP_PLIO_AF_API_KEY: ${{ secrets.VUE_APP_PLIO_AF_API_KEY }}
```
Comment (deepansh96) [inline .github/workflows/deploy_to_s3_prod.yml:37]: is this the same as the variable in `.env.example` above - `VUE_APP_API_KEY`?

File: docs/DEPLOYMENT.MD [hunk 10]
```diff
@@ -1,21 +1,25 @@
...
 5. If you choose to set up a SSL, do so at this step.
 6. Use a DNS provider of your choice. Give an appropriate CNAME and enter the target address. This would be the address of the newly created Cloudfront distribution.
 7. Add the following lines to the `scripts` object in `package.json`.
    ```sh
-    "deploy-staging": "vue-cli-service build --mode staging && aws --profile $YOUR_AWS_PROFILE --region $YOUR_AWS_REGION s3 sync ./dist s3://$S3_STAGING_BUCKET_NAME --delete --acl public-read",
-     "deploy": "vue-cli-service build && aws --profile $YOUR_AWS_PROFILE --region $YOUR_AWS_REGION s3 sync ./dist s3://$S3_PROD_BUCKET_NAME --delete --acl public-read"
-     ```
+    "deploy-staging": "vue-cli-service build --mode staging && aws --profile $YOUR_AWS_PROFILE --region $YOUR_AWS_REGION s3 sync ./dist s3://$S3_STAGING_BUCKET_NAME --delete --acl public-read"
+   ```
 8. To deploy your code on to staging, use `npm run deploy-staging/yarn deploy-staging`.
```
Comment (deepansh96) [inline docs/DEPLOYMENT.MD]:
```suggestion
8. To deploy your code on to staging, use `npm run deploy-staging` or `yarn deploy-staging`.
```

File: docs/ENV.MD [hunk 11]
```diff
@@ -1,17 +1,29 @@
...
-## VUE_APP_BASE_URL_PLIO=''
+## VUE_APP_BASE_URL_PLIO
 Base URL for the Plio link.
 
-## VUE_APP_AWS_SQS_ACCESS_KEY=''
+## VUE_APP_AWS_SQS_ACCESS_KEY
 AWS access key for access to AWS SQS.
 
-## VUE_APP_AWS_SQS_SECRET_ACCESS_KEY=''
+## VUE_APP_AWS_SQS_SECRET_KEY
```
Comment (deepansh96) [inline docs/ENV.MD:13]: I think you were right when you wrote "secret access key" before :p
Sorry for getting it changed
I just read the aws SQS client API docs and there they call this secret access key
Its okay for now we can change it later on to stay consistent

File: docs/INSTALLATION.MD [hunk 12]
```diff
@@ -4,36 +4,39 @@
 
 ### git-secrets
 git-secrets scans commits, commit messages, and --no-ff merges to prevent adding secrets into your git repositories.
-For installing, follow steps at https://github.com/awslabs/git-secrets#installing-git-secrets.
+For installing, follow steps at https://github.com/awsla bs/git-secrets#installing-git-secrets.
```
Comment (deepansh96) [inline docs/INSTALLATION.MD]: I think there's some issue with this link?

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy_to_s3_prod.yml (hunk 9 above)
Comment (sudheshna-donthineni) [reply .github/workflows/deploy_to_s3_prod.yml:37]: Yeah, Pritam wanted it to be more specific.

File: docs/ENV.MD (hunk 11 above)
Comment (sudheshna-donthineni) [reply docs/ENV.MD:13]: Yeah, will do it in the next PR.

File: src/components/OTP.vue [hunk 13]
```diff
@@ -378,10 +378,7 @@ export default {
 
     /** Function that verifies the OTP */
     async verifyOTP() {
-      const response = await OTPAuth.verifyOTP(
-        parseInt(this.phoneNumber),
-        parseInt(this.OTPCode)
-      );
+      const response = await OTPAuth.verifyOTP(parseInt(this.phoneNumber), this.OTPCode);
```
Comment (pritamps) [inline src/components/OTP.vue:381]: Why was this change needed?

**Review by pritamps — DISMISSED** [review]
Approved, but mention in a reply here why this change was needed

File: .env.example (hunk 8 above)
Comment (sudheshna-donthineni) [reply .env.example:8]: Plio AF API key

File: src/components/OTP.vue (hunk 13 above)
Comment (sudheshna-donthineni) [reply src/components/OTP.vue:381]: So by converting to int, the code was being sent to the service as a number. In that case, leading zeros were being trimmed and led to incorrect matching. Now, I'm just sending in as a string.
